### DISCUSSION TITLE
Alternative to using pywry as a tabling option

### DISCUSSION THREAD

**Discussion by caballerofelipe** [OpenBB-finance/OpenBB#6077] [Ideas]
I've seen some videos for OpenBB where they show tables being shown directly in the terminal. I've bee using OpenBB Terminal 3.4.2 and all tables are shown on an external window which isn't ideal for me. I believe seeing a chart in an external window is the way to go but for tables I'm not a fan.

I would like to be able to configure this as an option in the OpenBB Terminal (either using pywry or not). And maybe when showing the table I would like to have these configuration options:

- Which column to sort by.
- Either Ascending or Descending order.
- Limit the amount of data shown.
- Select columns.

If tables are too long, maybe it would also be cool to show data in something like Less like git does it... That way event if you have large tables you could scroll and once you close it you would return seamlessly to where you were typing.

**Comment by jmaslek** [comment]
In the terminal, if you do `/featflags/interactive`, that will show the data frames in your terminal using  rich

```
2024 Feb 15, 11:54  🦋 /stocks/disc/ $ gainers

                                                              Gainers                                                               
┏━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━┳━━━━━━━━┳━━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━┓
┃ Symbol ┃ Name                 ┃ Price (Intraday) ┃ Change ┃ % Change ┃ Volume  ┃ Avg Vol (3 month) ┃ Market Cap ┃ PE Ratio (TTM) ┃
┡━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━╇━━━━━━━━╇━━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━┩
│ PEGA   │ Pegasystems Inc.     │ 67.26            │ 16.55  │ +32.64%  │ 1.284M  │ 365114            │ 5.62B      │ nan            │
├────────┼──────────────────────┼──────────────────┼────────┼──────────┼─────────┼───────────────────┼────────────┼────────────────┤
│ FROG   │ JFrog Ltd.           │ 45.27            │ 8.18   │ +22.04%  │ 6.651M  │ 911234            │ 4.748B     │ nan            │
├────────┼──────────────────────┼──────────────────┼────────┼──────────┼─────────┼───────────────────┼────────────┼────────────────┤
│ SLVM   │ Sylvamo Corporation  │ 56.30            │ 10.78  │ +23.64%  │ 383358  │ 340736            │ 2.339B     │ 8.33           │
├────────┼──────────────────────┼──────────────────┼────────┼──────────┼─────────┼───────────────────┼────────────┼────────────────┤
│ SHAK   │ Shake Shack Inc.     │ 95.96            │ 17.89  │ +22.92%  │ 2.909M  │ 816885            │ 4.059B     │ 2399.00        │
├────────┼──────────────────────┼──────────────────┼────────┼──────────┼─────────┼───────────────────┼────────────┼────────────────┤
│ APP    │ AppLovin Corporation │ 57.01            │ 10.14  │ +21.64%  │ 10.363M │ 2.964M            │ 19.38B     │ 203.61         │
└────────┴──────────────────────┴──────────────────┴────────┴──────────┴─────────┴───────────────────┴────────────┴────────────────┘
```

**Comment by caballerofelipe** [comment]
Oh, that's cool, thanks for the tip!

What about...

> And maybe when showing the table I would like to have these configuration options:
>    - Which column to sort by.
>    - Either Ascending or Descending order.
>    - Limit the amount of data shown.
>    - Select columns.

Is that feasible? If so is there some docs on how to do it?

Also, is there a `less` command option?

**Reply by jmaslek** [reply]
So depending on the function, there can be the following flags:

- `--sortby` : which oclumn to sort by
- `--reverse` : flip ascending/descending
- `--limit` : how many rows to show

No ability to select columns, and unclear what you mean by `less`.  Are you implying --limit 1 to show just 1 row (which is less than the default)?

**Reply by caballerofelipe** [reply]
This is what I wrote in my original message:
> If tables are too long, maybe it would also be cool to show data in something like Less like git does it... That way event if you have large tables you could scroll and once you close it you would return seamlessly to where you were typing.

Sorry if that wasn't too clear... I was saying for large columns maybe the command less could be used. I think that's what git does in the terminal when there's something too long to show in one window view. They pipe the output to the command less and that way you can scroll and view all data inside terminal using the keyboard and `less` shortcuts. You can see what I mean if you type ´history|less´ in the terminal, you can see history with the up and down arrows in the keyboard (to exit less just press q).
